js/alert-box.js

```diff
@@ -14,28 +14,69 @@ bcpl.alertBox = (($) => {
 		sessionStorage.setItem('isAlertDismissed', true);
 	};
 
-	const init = () => {
+	const renderAlertBox = (alertData) => {
+		const alertsTemplateHtml = $('#alerts-template').html();
+		const $alertsTarget = $('#alerts-target');
+
+		if (alertsTemplateHtml && alertsTemplateHtml.length) {
+			const alertsTemplate = Handlebars.compile(alertsTemplateHtml);
+
+			if (alertData && alertData.IsEmergency) {
+				alertData.EmergencyClass = 'emergency'; // eslint-disable-line no-param-reassign
+			}
+
+			$alertsTarget.html(alertsTemplate({ alertData }));
+		}
+
+		displayNotificationBar(!alertData);
+	};
+
+	const getAlertDescription = (callback) => {
+		if (callback && typeof callback === 'function') {
+			$.ajax(CONSTANTS.baseApiUrl + CONSTANTS.shared.urls.alertNotification)
+				.then(
+					(data) => data ? callback(data, true) : callback(undefined, false),
+					() => callback(undefined, false)
+				);
+		} else {
+			console.error('A missing or invalid callback has been supplied.');
+		}
+	};
+
+	const hideNotificationBar = ($container) => {
+		$container.addClass('dismissed');
+		$container.show();
+	};
+
+	const displayNotificationBar = (shouldHide) => {
```

Comment:
It looks like ```hideNotificationBar``` could be called twice (lines 57 and 65)? Is that desired?

Reply:
Great catch! fixed.